wandb/sdk/launch/agent/agent.py

```diff
@@ -121,25 +127,77 @@ def finish_job_id(self, job_id: Union[str, int]) -> None:
     @staticmethod
     def _parse_runner_config(config: Dict[str, Any]) -> Dict[str, Any]:
         """
-            Parses a RunnersConfig block from the agent config. Returns a dictionary
-            of support runners in the form of:
-            {
-                kubernetes: {
-                    namespace: "wandb"
-                },
-                sagemaker: {
-
-                },
-                local-process: {
-                    labels: [ "gpu-pool" ]
-                },
-                local-container: {
-                    labels: [ "gpu-pool"]
-                }
+        Parses a RunnersConfig block from the agent config. Returns a dictionary
+        of support runners in the form of:
+        {
+            kubernetes: {
+                namespace: "wandb"
+            },
+            sagemaker: {
+
+            },
+            local-process: {
+                labels: [ "gpu-pool" ]
+            },
+            local-container: {
+                labels: [ "gpu-pool"]
             }
+        }
         """
         pass
 
+    def _get_valid_queues(self, entity: str) -> list[RunQueue]:
+        """
+        Given an entity, return a list of run queues associated with that entity.
+        """
+        QUERY = gql(
+            """
+            query GetRunQueuesByEntity($entityName: String!) {
+                runQueues(entityName: $entityName) {
+                    resource_config
+                    queue_id
+                    name
+                } 
+            }
+            """
+        )
+
+        run_queues = self._api.client.execute(QUERY, {"entityName": entity})
+        resource_configs = [json.loads(q["resourceConfig"]) for q in run_queues]
+        valid_queues = []
+
+        # can't wait for structural pattern matching...
+        for q, rc in zip(run_queues, resource_configs):
```

Comment:
do we need to split this out then rezip it up? coudln't we just check the `resourceConfig` key of the return run_queues here?